framework/test/jest/unit/specs/modules/chain/dpos/apply.spec.js

```diff
@@ -0,0 +1,466 @@
+		});
+	});
+
+	describe('Given block height is "1" (the genesis block)', () => {
+		let genesisBlock;
+		beforeEach(() => {
+			// Arrange
+			genesisBlock = {
+				height: 1,
+			};
+		});
+
+		it('should resolve with "false"', async () => {
+			// Act
+			const result = await dpos.apply(genesisBlock);
+
+			// Assert
+			expect(result).toBeFalse();
+		});
+
+		it('should NOT update "producedBlocks", "missedBlocks", "rewards", "fees", "votes"', async () => {
+			// Act
+			await dpos.apply(genesisBlock);
+
+			// Assert
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).not.toHaveBeenCalled();
+
+			expect(stubs.storage.entities.Account.update).not.toHaveBeenCalled();
+		});
+	});
+
+	it('should increase "producedBlocks" field by "1" for the generator delegate', async () => {
+		// Arrange
+		const block = {
+			height: 2,
+			generatorPublicKey: 'generatorPublicKey#RANDOM',
+		};
+
+		// Act
+		await dpos.apply(block, stubs.tx);
+
+		// Assert
+		expect(stubs.storage.entities.Account.increaseFieldBy).toHaveBeenCalledWith(
+			{ publicKey_eq: block.generatorPublicKey },
+			'producedBlocks',
+			'1',
+			stubs.tx,
+		);
+	});
+
+	describe('Given block is NOT the last block of the round', () => {
+		it('should NOT update "missedBlocks", "rewards", "fees", "votes"', async () => {
+			// Arrange
+			const block = {
+				height: 2,
+				generatorPublicKey: 'generatorPublicKey#RANDOM',
+			};
+
+			// Act
+			await dpos.apply(block, stubs.tx);
+
+			// Assert
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).toHaveBeenCalledTimes(1);
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).toHaveBeenCalledWith(
+				expect.any(Object),
+				'producedBlocks',
+				expect.any(String),
+				expect.anything(),
+			);
+
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).not.toHaveBeenCalledWith(expect.any(Object), 'missedBlocks');
+			expect(
+				stubs.storage.entities.Account.increaseFieldBy,
+			).not.toHaveBeenCalledWith(expect.any(Object), 'voteWeight');
+			expect(stubs.storage.entities.Account.update).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('Given block is the last block of the round', () => {
+		let lastBlockOfTheRound;
+		let feePerDelegate;
+		let rewardPerDelegate;
+		let totalFee;
+		let getTotalEarningsOfDelegate;
+		beforeEach(() => {
+			// Arrange
+			lastBlockOfTheRound = {
+				height: 202,
+				generatorPublicKey: delegateWhoForgedLast.publicKey,
+			};
+
+			when(stubs.storage.entities.Account.get)
+				.calledWith(
+					{
+						publicKey_in: delegatesWhoForged.map(({ publicKey }) => publicKey),
+					},
+					{},
+					stubs.tx,
+				)
+				.mockResolvedValue(delegatesWhoForged);
+
+			delegateAccounts.forEach(account => {
+				when(stubs.storage.entities.Account.get)
+					.calledWith({
+						publicKey_eq: account.publicKey,
+					})
+					.mockResolvedValue(account);
+			});
+
+			feePerDelegate = randomInt(10, 100);
```

Comment:
I remember this was intentionally added by @yatki for some reason and I would prefer not changing it without his consent.
Also, one could argue that It's not a random value. It's an arbitrary value :)

Reply:
Maybe @yatki can present his opinion on it later. If there is a bug which causes failure only on value 11 then this test may produce it but it will difficult to reproduce and will depend on chance. Anyway, we can keep it like this until @yatki responds.:)